testimony/__init__.py

```diff
@@ -92,190 +77,66 @@ def __init__(self, function_def, parent_class=None, testmodule=None):
-        unexpected_tags.update(class_unexpected_tags)
-        unexpected_tags.update(function_unexpected_tags)
-
-        for tag, value in tags.items():
-            if tag == 'bz':
-                tag = 'bugs'
-            if tag == 'assert':
-                tag = 'assertion'
-            if tag == 'type':
-                tag = 'test_type'
-            setattr(self, tag, value)
-        self.unexpected_tags = unexpected_tags
+        # Parse module, class and function docstrings. Every loop update the
+        # already defined tokens and invalid_tokens. Doing in the mentioned
+        # order will ensure that function docstrings have more priority over
+        # class and module docstring respectively.
+        docstrings = [
+            ast.get_docstring(self.module_def),
+            ast.get_docstring(self.parent_class_def),
+            self.docstring
+        ]
+        for docstring in docstrings:
+            if docstring and not isinstance(docstring, type(u'')):
+                docstring = docstring.decode('utf-8')
+            tokens, invalid_tokens = self.parser.parse(docstring)
+            self.tokens.update(tokens)
+            self.invalid_tokens.update(invalid_tokens)
 
         # Always use the first line of docstring as test case name
-        if self.test is None:
-            self.test = self.docstring.strip().split('\n')[0]
-
-    @property
-    def automated(self):
-        """Indicate if the test case is automated or not."""
-        if self.status and self.status.lower() == 'manual':
-            return False
-        return True
+        if self.tokens.get('test') is None:
+            self.tokens['test'] = self.docstring.strip().split('\n')[0]
 
     @property
     def has_valid_docstring(self):
-        """Returns ``True`` if at least feature and assert tags are defined.
-        """
-        return (
-            self.assertion is not None and
-            self.feature is not None and
-            self.test is not None
-        )
+        """Boolean indicating if docstring have the minimum tokens."""
+        return self.parser.validate_tokens(self.tokens.keys())
 
     def to_dict(self):
-        """Return the information in dict format.
-
-        This is a helper for converting to JSON.
-        """
-        return {
-            'assertion': self.assertion,
-            'automated': self.automated,
-            'bugs': self.bugs,
-            'feature': self.feature,
-            'name': self.name,
-            'parent_class': self.parent_class,
-            'setup': self.setup,
-            'skipped-lines': self.skipped_lines,
-            'status': self.status,
-            'steps': self.steps,
-            'tags': self.tags,
-            'test': self.test,
-            'test_type': self.test_type,
-            'unexpected-tags': self.unexpected_tags,
-        }
+        """Return tokens dictionary and include invalid-tokens as an item."""
+        values = self.tokens.copy()
```

Comment:
You have a point, initially I though on having the tokens keys directly on the return dict, but it seems to make more sense to have tokens and invalid-tokens keys.